lib/chef/provider/git.rb

```diff
@@ -137,28 +93,18 @@ def add_remotes
       end
 
       def clone
-        converge_by("clone from #{@new_resource.repository} into #{@new_resource.destination}") do
-          remote = @new_resource.remote
-
-          args = []
-          args << "-o #{remote}" unless remote == "origin"
-          args << "--depth #{@new_resource.depth}" if @new_resource.depth
-          args << "--no-single-branch" if @new_resource.depth && git_minor_version >= Gem::Version.new("1.7.10")
-
-          Chef::Log.info "#{@new_resource} cloning repo #{@new_resource.repository} to #{@new_resource.destination}"
-
-          clone_cmd = "git clone #{args.join(' ')} \"#{@new_resource.repository}\" \"#{@new_resource.destination}\""
-          shell_out!(clone_cmd, run_options)
+        converge_by("clone from #{@new_resource.repository} into #{cwd}") do
+          clone_by_advertized_ref
```

Comment:
should be `clone_by_advertised_ref` (oddly enough British and American English both agree here)